### ISSUE TITLE
Strange behavior if keyboard layout was changed before starting wezterm

### ISSUE THREAD

**Issue by ProspectPyxis** [wez/wezterm#5124]
### What Operating System(s) are you seeing this problem on?

Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

i3 version 4.23

### WezTerm version

20240226-174525-22424c32

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

Wezterm seems to have weird handling if the keyboard layout was changed before it's started.

The first GUI instance of wezterm doesn't seem to register the keyboard layout change and will always act as if it's the first/default layout. However, if a second wezterm instance is launched - either while the first is running or shortly after the first is closed - this issue disappears and both terminals register the layout change correctly. This issue is also fixed if another layout change is triggered while wezterm is running.

The "fixed" state is temporary, though; the issue comes back if wezterm is closed for a few seconds before being reopened.

### To Reproduce

1. Change your keyboard layout to anything other than the default set by your xkeyboard configuration.
2. Start wezterm.
3. Keystrokes incorrectly register as the first listed layout.
4. Launch a second wezterm instance.
5. Keystrokes now register correctly on both the first and second windows.

### Configuration

No config

### Expected Behavior

The keyboard layout should register correctly from the start.

### Logs

19:21:52.818  WARN   window::os::x11::connection > Unable to resolve appearance using xdg-desktop-portal: get_appearance.read_setting: Reading xdg-portal org.freedesktop.appearance color-scheme: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.portal.Desktop was not provided by any .service files

**Comment by ProspectPyxis** [comment]
Update on this after some testing - the behavior seems to be different while testing on nightly compared to stable, though that could also be because of the appimage packaging.

With the installed stable version, the "fixed" state persists until system reboot - not even logging out seems to replicate the bug here. Also, running the stable version just once "fixes" the appimage build as well.

**Comment by ProspectPyxis** [comment]
Alright, been a while and no idea what changed, but I finally switched to the latest nightly and this seems to be fixed - I'll close this issue.

**Comment by ProspectPyxis** [comment]
Ack, nevermind, looks like I just got lucky - the bug seems to be back. Though, the behavior is at least seeming consistent this time.

I'll open a new issue - I have some more details that probably belongs in the main issue post.
